RP Cloud Managed Connectors (#1577)
redpanda-data/documentation: docs/deploy/deployment-option/cloud/managed-connectors/create-jdbc-sink-connector.mdx

---
title: Create a JDBC Sink Connector
---

<head>
    <meta name="title" content="Create a JDBC Sink Connector | Redpanda Docs"/>
    <meta name="description" content="Use the Redpanda Cloud UI to create a JDBC Sink Connector."/>
    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/managed-connectors/create-jdbc-sink-connector/" />
</head>

> You can use a JDBC Sink connector to transfer structured data from Redpanda into

Reviewer: ```suggestion
You can use a JDBC Sink connector to export structured data from Redpanda to
```

   | ------------------------------------ | ----------- |
   | `Topics to export`             | Comma-separated list of the cluster topics you want to replicate. |
   | `JDBC URL`    | The database connection JDBC URL. |
   | `User`  | Name of the database user to be used when connecting to the database. |
   | `Password` | Password of the database user to be used when connecting to the database. |
   | `Kafka message key format`     | Format of the key in the Kafka topic. `BYTES` is the default. |
   | `Kafka message value format`   | Format of the value in the Kafka topic. `JSON` is the default. |
   | `Auto-create`  | When enabled, automatically creates the destination table (if it is missing) based on the record schema (issues a ``CREATE``). The default is disabled. |
   | `Max Tasks`                    | Maximum number of tasks to use for this connector. The default is `1`. Each task replicates exclusive set of partitions assigned to it. |
   | `Connector name`               | Globally-unique name to use for this connector. |
4. Click **Next**. Review the connector properties specified, then click **Create**.

### Advanced JDBC Sink connector configuration

> In most instances, the preceding basic configuration properties are sufficient.
> However, if you require any additional property settings, then specify any of the following

Reviewer: ```suggestion
If you require additional property settings, then specify any of the following
```

| `Maximum retries` | The maximum number of times to retry on errors before failing the task. The default is `10`. |
| `Retry backoff (ms)` | The time in milliseconds to wait before a retry attempt is made following an error. The default is `3000`. |
| `Database dialect` | The name of the database dialect that should be used for this connector. By default. the connector automatically determines the dialect based upon the JDBC connection URL. Use if you want to override that behavior and specify a specific dialect. |

## Map data

Use the appropriate key or value converter (input data format) for your data as follows:

- Use the default `Kafka message value format=JSON` property in your configuration.
- Topics should contain data in JSON format with a defined JSON schema. For example:
  ```json
  {
     "schema": {
       "type": "struct",
       "fields": [

>          // fields

Reviewer: ```suggestion
```

       ]
     },
     "payload": {

>          // payload

Reviewer: ```suggestion
```

     }
  }
  ```

## Test the connection

> After the connector is created, check to ensure that:

Reviewer: ```suggestion
After the connector is created, ensure that:
```
